src/Microsoft.Sbom.Targets/GenerateSbomTask.cs

```diff
@@ -24,52 +27,62 @@ namespace Microsoft.Sbom.Targets;
 /// <summary>
 /// MSBuild task for generating SBOMs from build output.
 /// </summary>
-public partial class GenerateSbom : Task
+public partial class GenerateSbom : Task, ICancelableTask
```

Comment:
Added `CancelableTask` support to help clean up - though many of the task-using calls lower in the stack don't support CancellationTokens. The MSBuild side is now ready to handle them if/when they get support - right now you get a pretty nasty build hang so eventually it would be nice to fix that.